src/api/IBMi.ts

```diff
@@ -1231,4 +1257,55 @@ export default class IBMi {
       return name.toLocaleUpperCase();
     }
   }
+
+
+  /**
+   * Run SQL statements.
+   * Each statement must be separated by a semi-colon and a new line (i.e. ;\n).
+   * If a statement starts with @, it will be run as a CL command.
+   *
+   * @param statements
+   * @returns a Result set
+   */
+  async runSQL(statements: string, opts: {userCcsid?: number} = {}): Promise<Tools.DB2Row[]> {
+    const { 'QZDFMDB2.PGM': QZDFMDB2 } = this.remoteFeatures;
+
+    if (QZDFMDB2) {
+      const ccsidDetail = this.getEncoding();
+      const useCcsid = opts.userCcsid || (ccsidDetail.fallback && !ccsidDetail.invalid ? ccsidDetail.ccsid : undefined);
+      const possibleChangeCommand = (useCcsid ? `@CHGJOB CCSID(${useCcsid});\n` : '');
+
+      const output = await this.sendCommand({
+        command: `LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"`,
+        stdin: Tools.fixSQL(`${possibleChangeCommand}${statements}`)
+      })
+
+      if (output.stdout) {
+        return Tools.db2Parse(output.stdout);
+      } else {
+        throw new Error(`There was an error running the SQL statement.`);
+      }
+
+    } else {
+      throw new Error(`There is no way to run SQL on this system.`);
+    }
+  }
+
+  getEncoding() {
+    const fallback = ((this.runtimeCcsid < 1 || this.runtimeCcsid === 65535) && this.userDefaultCCSID > 0 ? true : false);
```

Comment:
As much as I love ternaries, this is already a boolean expression 😅
```suggestion
    const fallback = (this.runtimeCcsid < 1 || this.runtimeCcsid === 65535) && this.userDefaultCCSID > 0;
```